src/Raven.Server/Utils/RavenCli.cs

```diff
@@ -60,78 +64,93 @@ internal static class RavenCli
                 }
             }
             if (test == false)
-                Console.Write(msg);
+                writer.Write(msg);
         };
-        private static List<string> _promptArgs = new List<string> { "ravendb" };
+
+        private List<string> _promptArgs = new List<string> { "ravendb" };
+        private TextWriter _writer;
+        private TextReader _reader;
 
 
         private class SingleAction
         {
             public int NumOfArgs;
-            public Func<List<string>, bool> DelegateFync;
+            public Func<List<string>, TextWriter, TextReader, bool, RavenServer, bool> DelegateFync;
+            public bool Experimental { get; set; }
         }
 
-        private static bool CommandQuit(List<string> args)
+        private static bool CommandQuit(List<string> args, TextWriter writer, TextReader reader, bool consoleColoring, RavenServer server)
         {
-            Console.ResetColor();
-            Console.WriteLine();
-            Console.Write("Are you sure you want to quit the server ? [y/N] : ");
-            Console.Out.Flush();
+            ResetColor(consoleColoring);
+            writer.WriteLine();
+            writer.Write("Are you sure you want to quit the server ? [y/N] : ");
+            writer.Flush();
 
-            var k = Console.ReadKey();
-            Console.Out.Flush();
+            var k = ReadKey(consoleColoring, reader);
+            writer.Flush();
 
-            Console.WriteLine();
-            if (k.KeyChar.Equals('Y') ||
-                k.KeyChar.Equals('y'))
-                return true;
-            return false;
+            writer.WriteLine();
+            return char.ToLower(k).Equals('y');
         }
 
-        private static bool CommandResetServer(List<string> args)
+        private static char ReadKey(bool consoleColoring, TextReader reader)
         {
-            Console.ResetColor();
-            Console.WriteLine();
-            Console.Write("Are you sure you want to reset the server ? [y/N] : ");
-            Console.Out.Flush();
+            if (consoleColoring)
+                return Console.ReadKey().KeyChar;
 
-            var k = Console.ReadKey();
-            Console.Out.Flush();
+            var c = new char[1];
+            while (reader.Read(c, 0, 1) < 1)
+            {
 
-            Console.WriteLine();
-            if (k.KeyChar.Equals('Y') ||
-                k.KeyChar.Equals('y'))
-                return true;
-            return false;
+            }
+            return c[0];
         }
 
-        private static bool CommandStats(List<string> args)
+        private string ReadLine()
+        {
+            return _consoleColoring ? Console.ReadLine() : _reader.ReadLine();
+        }
+
+        private static bool CommandResetServer(List<string> args, TextWriter writer, TextReader reader, bool consoleColoring, RavenServer server)
```

Comment:
You are passing `consoleColoring` all over the place, just make it a field and use instance methods